staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/marshal_test.go

```diff
@@ -54,9 +59,107 @@ func TestJSONSchemaPropsOrBoolUnmarshalJSON(t *testing.T) {
 			t.Errorf("Failed to unmarshal input '%v': expected %+v, got %+v", c.input, c.result, result)
 		}
 	}
+
+	t.Run("null", func(t *testing.T) {
+		var actual JSONSchemaPropsOrBool
```

Comment:
The JSON behavior actually resets the receiver to the zero value, so this should be initialized with some dummy data, then verified that it is the zero value after unmarshal.